cxx4opencl/address_spaces.txt

```diff
@@ -229,6 +315,31 @@ __global const int& f(__global float &ref) {
 }
 ----------
 
-==== Initialization of local and constant address space objects
-
-TODO
+==== Construction, initialization and destruction
+
+Construction, initialization and destruction of objects in `__private`
+and `__global` address space follows the general principals of {cpp}. For
+the program scope objects the implementation (i.e. compiler) defines an ABI
+format for initialization and destruction of global objects before/after
+all kernels are enqueued.
+
+Objects in `__local` address space can not have initializers in
+declarations and therefore a constructor can not be called. All objects
+created in the local address space have undefined state at the beginning.
```

Comment:
Could say something more  precise that "beginning".   I suspect that it means "at the start of kernel execution".

Reply:
Yes, or at the point of declaration... not sure which one is more precise... or perhaps there is no difference...

Reply:
Changed in https://github.com/KhronosGroup/OpenCL-Docs/pull/263/commits/adeb8aae4ab18510f35a84cb6db53c4bbd716d32